torch_xla_py/xla_dist.py

```diff
@@ -402,3 +409,243 @@ def get_cluster(self):
     cluster.validate()
     return cluster
 
+
+class DistributedExecutor(object):
+
+  SCRIPT_PATH_TMPL = '/tmp/dist_training_ptxla_{worker}.sh'
+  MASTER_IDX = 0
+  MESH_SERVICE_PORT = 8477  # Use single port to disallow concurrent runs
+  DOCKER_RUN = 'docker run'
+  DIST_ENV_VARS = [
+      'XRT_TPU_CONFIG', 'XRT_LOCAL_WORKER', 'XRT_MESH_SERVICE_ADDRESS'
+  ]
+
+  @staticmethod
+  def _parse_container_name(cmd):
+    for key in ['--name=', '--name']:
+      if key in cmd:
+        # ex. "docker run --name=pytorch image"
+        idx = cmd.index(key)
+        post_name = cmd[idx:].split(key)[1]  # ex. "pytorch image"
+        return post_name.split()[0]  # ex. "pytorch"
+
+    return None
+
+  def __init__(self, cluster):
+    self._cluster = cluster
+    logging.basicConfig(
+        format='%(asctime)-12s %(clientip)s [%(ordinal)s] %(message)s',
+        datefmt='%Y-%m-%d %H:%M:%S',
+        level=logging.DEBUG)
+    self.logger = logging.getLogger('DistributedExecutor')
+    self.container_name = None
+    self.is_docker = False
+
+  def _stream_logs(self, process, client_worker):
+    client_ip = client_worker._internal_ip
+    ordinal = self._cluster._client_workers.index(client_worker)
+
+    def _stream_stdout():
+      for stdout in iter(process.stdout.readline, b''):
+        self.logger.info(stdout.decode('utf-8').rstrip('\n'),
+                         extra={'clientip': client_ip, 'ordinal': ordinal})
+
+    def _stream_stderr():
+      for stderr in iter(process.stderr.readline, b''):
+        self.logger.error(stderr.decode('utf-8').rstrip('\n'),
+                         extra={'clientip': client_ip, 'ordinal': ordinal})
+
+    stdout = threading.Thread(target=_stream_stdout)
+    stdout.daemon = True
+    stdout.start()
+    stderr = threading.Thread(target=_stream_stderr)
+    stderr.daemon = True
+    stderr.start()
+    stdout.join()
+    stderr.join()
+
+  def _add_docker_vars(self, cmd):
+    # Get or set container name to explicitly kill at main thread ctrl+c
+    self.container_name = self._parse_container_name(cmd)
+    container_name_cmd = ''
+    if self.container_name is None:
+      # Name not set by user
+      self.container_name = 'pytorchtpudistrunner'
+      container_name_cmd = '--name={}'.format(self.container_name)
+
+    docker_export_cmd = ' -e '.join([''] + DistributedExecutor.DIST_ENV_VARS)
```

Comment:
`DIST_ENV_VARS` seems to be a list of keys only, doesn't have the values. Shouldn't we add values to the command?